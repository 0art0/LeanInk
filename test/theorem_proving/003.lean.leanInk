[{"contents":
  "theorem test (p q : Prop) (hp : p) (hq : q) : p ∧ q ∧ p := by\n  ",
  "_type": "text"},
 {"messages": [],
  "goals":
  [{"name": "",
    "hypotheses": [],
    "conclusion": "\np q : Prop hp : p hq : q ⊢ p ∧ q ∧ p",
    "_type": "goal"}],
  "contents": "apply And.intro",
  "_type": "sentence"},
 {"contents": "\n  ", "_type": "text"},
 {"messages": [],
  "goals":
  [{"name": "",
    "hypotheses": [],
    "conclusion":
    "\ncase left\np q : Prop\nhp : p\nhq : q\n⊢ p\ncase right p q : Prop hp : p hq : q ⊢ q ∧ p",
    "_type": "goal"}],
  "contents": "exact hp",
  "_type": "sentence"},
 {"contents": "\n  ", "_type": "text"},
 {"messages": [],
  "goals":
  [{"name": "",
    "hypotheses": [],
    "conclusion": "\ncase right p q : Prop hp : p hq : q ⊢ q ∧ p",
    "_type": "goal"}],
  "contents": "apply And.intro",
  "_type": "sentence"},
 {"contents": "\n  ", "_type": "text"},
 {"messages": [],
  "goals":
  [{"name": "",
    "hypotheses": [],
    "conclusion":
    "\ncase right.left\np q : Prop\nhp : p\nhq : q\n⊢ q\ncase right.right p q : Prop hp : p hq : q ⊢ p",
    "_type": "goal"}],
  "contents": "exact hq",
  "_type": "sentence"},
 {"contents": "\n  ", "_type": "text"},
 {"messages": [],
  "goals":
  [{"name": "",
    "hypotheses": [],
    "conclusion": "\ncase right.right p q : Prop hp : p hq : q ⊢ p",
    "_type": "goal"}],
  "contents": "exact hp",
  "_type": "sentence"},
 {"contents": "", "_type": "text"}]