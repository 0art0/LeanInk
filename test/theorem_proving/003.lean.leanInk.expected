[{"contents":
  "theorem test (p q : Prop) (hp : p) (hq : q) : p ∧ q ∧ p := by\n  ",
  "_type": "text"},
 {"messages": [],
  "goals":
  [{"name": "left",
    "hypotheses":
    [{"type": "Prop", "names": ["p", "q"], "body": "", "_type": "hypothesis"},
     {"type": "p", "names": ["hp"], "body": "", "_type": "hypothesis"},
     {"type": "q", "names": ["hq"], "body": "", "_type": "hypothesis"}],
    "conclusion": "p",
    "_type": "goal"},
   {"name": "right",
    "hypotheses":
    [{"type": "Prop", "names": ["p", "q"], "body": "", "_type": "hypothesis"},
     {"type": "p", "names": ["hp"], "body": "", "_type": "hypothesis"},
     {"type": "q", "names": ["hq"], "body": "", "_type": "hypothesis"}],
    "conclusion": "q ∧ p",
    "_type": "goal"}],
  "contents": "apply And.intro",
  "_type": "sentence"},
 {"contents": "\n  ", "_type": "text"},
 {"messages": [],
  "goals":
  [{"name": "right",
    "hypotheses":
    [{"type": "Prop", "names": ["p", "q"], "body": "", "_type": "hypothesis"},
     {"type": "p", "names": ["hp"], "body": "", "_type": "hypothesis"},
     {"type": "q", "names": ["hq"], "body": "", "_type": "hypothesis"}],
    "conclusion": "q ∧ p",
    "_type": "goal"}],
  "contents": "exact hp",
  "_type": "sentence"},
 {"contents": "\n  ", "_type": "text"},
 {"messages": [],
  "goals":
  [{"name": "right.left",
    "hypotheses":
    [{"type": "Prop", "names": ["p", "q"], "body": "", "_type": "hypothesis"},
     {"type": "p", "names": ["hp"], "body": "", "_type": "hypothesis"},
     {"type": "q", "names": ["hq"], "body": "", "_type": "hypothesis"}],
    "conclusion": "q",
    "_type": "goal"},
   {"name": "right.right",
    "hypotheses":
    [{"type": "Prop", "names": ["p", "q"], "body": "", "_type": "hypothesis"},
     {"type": "p", "names": ["hp"], "body": "", "_type": "hypothesis"},
     {"type": "q", "names": ["hq"], "body": "", "_type": "hypothesis"}],
    "conclusion": "p",
    "_type": "goal"}],
  "contents": "apply And.intro",
  "_type": "sentence"},
 {"contents": "\n  ", "_type": "text"},
 {"messages": [],
  "goals":
  [{"name": "right.right",
    "hypotheses":
    [{"type": "Prop", "names": ["p", "q"], "body": "", "_type": "hypothesis"},
     {"type": "p", "names": ["hp"], "body": "", "_type": "hypothesis"},
     {"type": "q", "names": ["hq"], "body": "", "_type": "hypothesis"}],
    "conclusion": "p",
    "_type": "goal"}],
  "contents": "exact hq",
  "_type": "sentence"},
 {"contents": "\n  exact hp", "_type": "text"}]